packages/bitcore-node/scripts/syncTxnAndCoinsWallets.js

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/env node
+'use scrict';
+
+/**
+ * This script fixes an issue where the wallets are in Coin.wallets but not Transaction.wallets in the db
+ */
+
+const fs = require('fs');
+const { Storage } = require('../build/src/services/storage');
+const { TransactionStorage } = require('../build/src/models/transaction');
+const { CoinStorage } = require('../build/src/models/coin');
+
+function usage(err) {
+  if (err) {
+    console.log(err);
+    console.log('');
+  }
+  console.log('Usage: ./fixTxnToWalletMarriage.js [options] <chain> <network> <startBlockHeight> <endBlockHeight>');
+  console.log('');
+  console.log('  --chain <value>          BTC, BCH, ETH, etc.');
+  console.log('  --network <value>        mainnet, testnet, or regtest');
+  console.log('  --startHeight <value>    Block height to start at (inclusive)');
+  console.log('  --endHeight <value>      Block height to end at (inclusive)');
+  console.log('  --no-dry                 Run it for real');
+  console.log('  --no-log-file            Output to console instead of writing to a log file');
+  process.exit();
+};
+
+const chain = process.argv.find(a => a == '--chain') ? process.argv[process.argv.indexOf('--chain') + 1] : undefined;
+const network = process.argv.find(a => a == '--network') ? process.argv[process.argv.indexOf('--network') + 1] : undefined;
+const startBlockHeight = process.argv.find(a => a == '--startHeight') ? Number(process.argv[process.argv.indexOf('--startHeight') + 1]) : undefined;
+const endBlockHeight = process.argv.find(a => a == '--endHeight') ? Number(process.argv[process.argv.indexOf('--endHeight') + 1]) : undefined;
+const dryRun = process.argv.find(a => a == '--no-dry') ? false : true;
+const noLogFile = process.argv.find(a => a == '--no-log-file') ? true : false;
+
+if (process.argv.includes('--help') || process.argv.includes(['-h'])) {
+  usage();
+}
+
+if (!chain) {
+  usage('Invalid chain: ' + chain);
+}
+
+if (!network) {
+  usage('Invalid network: ' + network);
+}
+
+if (!startBlockHeight || isNaN(startBlockHeight)) {
+  usage('Invalid startBlockHeight: ' + startBlockHeight);
+}
+
+if (!endBlockHeight || isNaN(endBlockHeight)) {
+  usage('Invalid endBlockHeight: ' + endBlockHeight);
+}
+
+if (!dryRun) {
+  console.log('RUNNING FOR REAL. DATA WILL BE MODIFIED IN THE DATABASE');
+}
+
+console.log('Connecting to database...');
+
+Storage.start({ dbName: 'bitcore_mocha_scripts' })
```

Comment:
I see the point in leaving the test db here for when the test runs but I am curious if there is a better way to handle this. I think at a minimum you should leave a comment at the top of the script indicating that the db name needs to be changed to run against prod. You could also make this a variable defined up top so that it is easily changed without having to dive into the script.

Reply:
D'oh! I kept seeing this while I was testing and reminding myself to remove it before committing. ...I forgot :-1: